client/http/metric.go

```diff
@@ -16,7 +16,9 @@ func MeasureHeartbeats(ctx context.Context, c []client.Client) *HealthMetrics {
 		next:    0,
 		clients: c,
 	}
-	go m.startObserve(ctx)
+	if len(c) > 0 {
```

Comment:
should it warn, log, die when there are no clients? Would this be an abnormal thing?

Reply:
based on our use of this in the client binary, it isn't a warning - it will happen in normal operation when the client is configured to use grpc or gossip randomness and does not have a valid http upstream endpoint